tests/test_bigbio.py

```diff
@@ -0,0 +1,660 @@
+                        ):
+
+                            event_id = event["id"]
+                            errors.append(
+                                f"Example:{example_id} - event:{event_id} "
+                                + msg
+                            )
+
+        if len(errors) > 0:
+            self.fail(msg="\n".join(errors) + OFFSET_ERROR_MSG)
+
+    def test_coref_ids(self):
+        """
+        Verify that coreferences ids are entities
+
+        from `examples/test_n2c2_2011_coref.py`
+        """  # noqa
+        log.info("KB ONLY: Checking coref offsets")
+        for split in self.dataset_bigbio:
+
+            if "coreferences" in self.dataset_bigbio[split].features:
+
+                for example in self.dataset_bigbio[split]:
+                    entity_lookup = {
+                        ent["id"]: ent for ent in example["entities"]
+                    }
+
+                    # check all coref entity ids are in entity lookup
+                    for coref in example["coreferences"]:
+                        for entity_id in coref["entity_ids"]:
+                            assert entity_id in entity_lookup
+
+    def test_name(self):
+        """
+        Checks if the dataloader script is in the right nested structure.
+        All dataloader scripts must be of the following format:
+
+        biomedical/biomeddatasets/<name_of_dataset>/<name_of_dataloader_script>
+        """  # noqa
+        datafolder = Path("biomeddatasets/") / self.NAME
+        datascript = datafolder / self.NAME + ".py"
+        self.assertTrue(datafolder.exists(), "Folder not named " + self.NAME)
+
+        self.assertTrue(
+            datascript.exists(),
+            "Script not named " + self.NAME + ".py",
+        )
+
+    def test_schema(self, task: str):
+        """Search supported tasks within a dataset and verify big-bio schema"""
+
+        mapped_task = self._test_task_exists(task)
+
+        if mapped_task == "kb":
+            log.info("NER/NED/RE/Event Extraction/Coref Task")
+
+            schema = kb_features
+
+            # TODO - change this
+            opt_keys = [
+                "entities",
+                "events",
+                "coreferences",
+                "relations",
+            ]
+
+            needed_keys = [
+                key for key in schema.keys() if key not in opt_keys
+            ]
+
+            for split in self.dataset_bigbio.keys():
+                example = self.dataset_bigbio[split][0]
+
+                # Check for mandatory keys
+                mandatory_keys = all(
+                    [
+                        key in example
+                        for key in schema.keys()
+                        if key not in opt_keys
+                    ]
+                )
+                self.assertTrue(
+                    mandatory_keys,
+                    "id/document_id/passages missing from bigbio view",
+                )
```

Comment:
I sometimes handle this kind of thing with set subtraction ... if you like this formulation feel free to use it, 
```
missing_keys = set(needed_keys) - set(example.keys())
self.assertTrue(
    len(missing_keys) == 0,
   f"{missing_keys} are missing from bigbio view",
)
```